3주차/FE/minjaemoon/src/App.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import { MdAddCircle } from "react-icons/md";
+import "./App.css";
+import Template from "./components/Template";
+import TodoList from "./components/TodoList";
+import TodoInsert from "./components/TodoInsert";
+
+let nextId = 4;
+
+const App = () => {
+  const [selectedTodo, setSelectedTodo] = useState(null);
+  const [insertToggle, setInsertToggle] = useState(false);
+  const [todos, setTodos] = useState([
+    {
+      id: 1,
+      text: "할일 1",
+      checked: true
+    },
+    {
+      id: 2,
+      text: "할일 2",
+      checked: false
+    },
+    {
+      id: 3,
+      text: "할일 3",
+      checked: true
+    }
+  ]);
+
+  const onInsertToggle = () => {
+    if (selectedTodo) {
+      setSelectedTodo(null);
+    }
+    setInsertToggle(prev => !prev);
+  };
+  const onInsertTodo = (text) => {
+    if (text === "" ) {
+      return alert('할 일을 입력해주세요.');
+    } else {
+      const todo = {
+        id: nextId,
+        text, 
+        checked: false
+      }
+      setTodos(todos => todos.concat(todo));
+      nextId++;
+    };
+  };
+  
+  const onCheckToggle = id => {
+    setTodos(todos => 
+      todos.map(todo => 
+        todo.id == id ? {...todo, checked: !todo.checked} : todo  
+      )
+    );
```

Comment:
아래와 같은 방법으로 작성해보시는 방법도 있습니다!
setTodos를 이용하지 않고도 같은 결과를 만들어낼 수 있습니다!
```suggestion
    todos.filter((todo) => todo.id === id).forEach((todo) => {todo.checked = !todo.checked})
```